tenpy/simulations/time_evolution.py

```diff
@@ -74,3 +80,347 @@ def final_measurements(self):
+        return super().prepare_results_for_save()
+
+    def get_mps_environment(self):
+        return MPSEnvironment(self.psi_ground_state, self.psi)
+
+    def m_spectral_function(self, results, psi, model, simulation, **kwargs):
+        """Calculate the overlap :math:`<psi_0| e^{iHt} op2^j e^{-iHt} op1_idx |psi_0>` between
+        op1 at MPS position idx and op2 at the MPS position j"""
+        self.logger.info("calling m_spectral_function")
+        operator_t = self.operator_t
+        env = self.get_mps_environment()  # custom method for subclass Experimental
+        # TODO: get better naming convention, store this in dict ?
+        if isinstance(operator_t, list):
+            for i, op in enumerate(operator_t):
+                if isinstance(op, str):
+                    results[f'spectral_function_t_{op}'] = self._m_spectral_function_op(env, op)
+                else:
+                    results[f'spectral_function_t_{i}'] = self._m_spectral_function_op(env, op)
+        else:
+            if isinstance(operator_t, str):
+                results[f'spectral_function_t_{operator_t}'] = self._m_spectral_function_op(env, operator_t)
+            else:
+                results[f'spectral_function_t'] = self._m_spectral_function_op(env, operator_t)
+
+    def _m_spectral_function_op(self, env: MPSEnvironment, op) -> np.ndarray:
+        """Calculate the overlap of <psi| op_j |phi>, where |phi> = e^{-iHt} op1_idx |psi_0>
+        (the time evolved state after op1 was applied at MPS position idx) and
+        <psi| is either <psi_0| e^{iHt} (if evolve_bra is True) or e^{i E_0 t} <psi| (if evolve_bra is False).
+
+        Returns
+        ----------
+        spectral_function_t : 1D array
+                              representing <psi_0| e^{iHt} op2^i_j e^{-iHt} op1_idx |psi_0>
+                              where op2^i is the i-th operator given in the list [op2^1, op2^2, ..., op2^N]
+                              and spectral_function_t[j] corresponds to this overlap at MPS site j at time t
+        """
+        # TODO: case dependent if op needs JW string
+        if self.addJW is False:
+            spectral_function_t = env.expectation_value(op)
+        else:
+            spectral_function_t = list()
+            for i in range(self.psi.L):
+                term_list, i0, _ = env._term_to_ops_list([('Id', 0), (op, i)], True)
+                # this generates a list from left to right
+                # ["JW", "JW", ... "JW", "op (at idx)"], the problem is, that _term_to_ops_list does not generate
+                # a JW string for one operator, therefore insert Id at idx 0.
+                assert i0 == 0  # make sure to really start on the left site
+                spectral_function_t.append(env.expectation_value_multi_sites(term_list, i0))
+                # TODO: change when :meth:`expectation_value` of :class:`MPSEnvironment` automatically handles JW-string
```

Comment:
Yes, again, we should implement #282 and adjust code here. (saying this again here to get another cross-reference in the github issue).